lib/src/web/database.dart

```diff
@@ -0,0 +1,282 @@
+import 'dart:async';
+import 'dart:js_interop';
+
+import 'package:sqlite3/common.dart';
+import 'package:sqlite3_web/sqlite3_web.dart';
+import 'package:sqlite_async/mutex.dart';
+import 'package:sqlite_async/sqlite_async.dart';
+import 'package:sqlite_async/src/common/sqlite_database.dart';
+import 'package:sqlite_async/src/sqlite_connection.dart';
+import 'package:sqlite_async/src/sqlite_queries.dart';
+import 'package:sqlite_async/src/update_notification.dart';
+import 'package:sqlite_async/src/utils/shared_utils.dart';
+import 'protocol.dart';
+
+class WebDatabase
+    with SqliteQueries, SqliteDatabaseMixin
+    implements SqliteDatabase {
+  final Database _database;
+  final Mutex? _mutex;
+
+  @override
+  bool closed = false;
+
+  WebDatabase(this._database, this._mutex);
+
+  @override
+  Future<void> close() async {
+    await _database.dispose();
+    closed = true;
+  }
+
+  @override
+  Future<bool> getAutoCommit() async {
+    final response = await _database.customRequest(
+        CustomDatabaseMessage(CustomDatabaseMessageKind.getAutoCommit));
+    return (response as JSBoolean?)?.toDart ?? false;
+  }
+
+  @override
+  Future<void> initialize() {
+    return Future.value();
+  }
+
+  @override
+  Future<void> get isInitialized => initialize();
+
+  @override
+
```

Comment:
Yes it works, this is also how the Dart formatter formats this if there is a comment for the method. This can be tested by removing the `@override`: the linter should give an error if it's removed and the error should go away even if there is a space.